### PR TITLE
rgw:modify command stucking when operating radosgw-admin metadata list user

### PR DESCRIPTION
**Pull request by liupeiyang**
when no user exists,the command stucks by operating radosgw-admin metadata list user .It seems all right now

Signed-off-by:  Peiyang Liu  <liu.peiyang@h3c.com>

### CODE REVIEW

**Comment by liupeiyang** [comment]
@yehudasa  what happed about the  commit 7032? All checks have failed?

**Comment by liupeiyang** [comment]
@oritwas could you look at it? #7032

File: src/rgw/rgw_user.cc [hunk 1]
```diff
@@ -2628,8 +2628,13 @@ class RGWUserMetadataHandler : public RGWMetadataHandler {
 
     int ret = store->list_raw_objects(store->zone.user_uid_pool, no_filter,
                                       max, info->ctx, unfiltered_keys, truncated);
-    if (ret < 0)
+    if (ret < 0 && ret != -ENOENT)
       return ret;
+    if (ret == -ENOENT) {
+      if (truncated)
+        *truncated = false;
+      return 0;
```
Comment (oritwas) [inline src/rgw/rgw_user.cc]: shouldn't we return here -ENOENT?

**Comment by liupeiyang** [comment]
@oritwas it should be better. it could be merge after modifying ,Is that so?
<pre> 
 if (ret < 0 && ret != -ENOENT)
        return ret;		    
 if (ret == -ENOENT) {
       if (truncated)
          *truncated = false;
       return -ENOENT;
</pre>

**Comment by liupeiyang** [comment]
I have revised the commit @oritwas

**Pull request merged**